OCPBUGS#11363: Expose the node port for the application (#58232)
openshift/openshift-docs: modules/nw-exposing-service.adoc

endif::nodeport[]
ifdef::nodeport[]

> . To expose a node port for the application, modify the custom resource definition (CRD) of a service by entering by the following command:

Reviewer: ```suggestion
. To expose a node port for the application, modify the custom resource definition (CRD) of a service by entering the following command:
```

+
[source,terminal]
----
$ oc edit svc <service_name>
----
+
.Example output
[source,yaml]
----
spec:
  ports:
  - name: 8443-tcp
    nodePort: 30327 <1>
    port: 8443
    protocol: TCP
    targetPort: 8443
  sessionAffinity: None
  type: NodePort <2>
----

> <1> Optional: Specify the node port range for the application. {product-title} automatically selects an available port in the `30000-32767` range.

Reviewer: ```suggestion
<1> Optional: Specify the node port range for the application. By default, {product-title} selects an available port in the `30000-32767` range.
```

Reviewer: And is a range something that we can insert into the YAML? If not, consider:

```suggestion
<1> Optional: Specify the node port range for the application. By default {product-title}, selects an available port in the 30000-32767 range.
```

Monospace on the dash implies that it is part of code or a configuration file.

> <2> Defines the service type.

Reviewer: Parallel construction.
```suggestion
<2> Define the service type.
```

> . Verify that the service node port is updated with the range `30327`:

Reviewer 2: ```suggestion
. Verify that the service node port is updated with a port in the correct range, for example, `30327`:
```

Reviewer 2: Suggesting this addition because it's automatically selecting a port, which could be something other than 30327. I think it just makes it a bit more clear. ("correct" may not be the best word here - maybe "appropriate" or "valid" would be better)

Author: Thank you, Kerry. 
Instead of saying the "correct" range, IMO we can use the actual range here- `30000-32767`. So updated with the range `30000-32767`.

Author: Thanks for pointing that out

> . Verify that the service node port is updated with a port in the `30000-32767` range, for example, `30327`:

Reviewer: This could be a verification step.

> . Verify that the service node port is updated with a port in the `30000-32767` range, for example, `30327`:

Reviewer: ```suggestion
. Verify that the service node port is updated with a port in the `30000-32767` range. For example, `30327`:
```

+
[source,terminal]
----
$ oc get svc
----
+
.Example output
[source,terminal]
----
NAME    TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)                         AGE

> httpd   NodePort   172.xx.xx.xx    <none>        8080:30546/TCP,8443:30327/TCP   109s

Reviewer 3: This needs updated too, remove " 8080:30546/TCP,"

endif::nodeport[]
ifdef::nodeport[]
. To expose a node port application, modify the custom resource definition of a service by entering by the following command:
+
[source,terminal]
----
$ oc edit svc <service_name>
----
+
.Example output
[source,terminal]
----
spec:
  ...
  ports:
  - name: 8080-tcp
  nodePort: 30546
  port: 8080
  protocol: TCP
  targetPort: 8080
  - name: 8443-tcp

>   nodePort: 30327 <1>

Reviewer 3: It seems the indentation is not correct for this yaml, please refer to the  bug.

Author: Updated the identation

----
$ oc delete svc nodejs-ex
----

.Verification

> * Verify that the service node port is updated with a port in the `30000-32767` range. For example, `30327`:
> +
> [source,terminal]
> ----
> $ oc get svc
> ----
> +
> .Example output
> [source,terminal]
> ----
> NAME    TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
> httpd   NodePort   172.xx.xx.xx    <none>        8443:30327/TCP   109s
> ----

Reviewer: Suggestion that includes the "enter the following command" language:
```suggestion
* To check that the service node port is updated with a port in the `30000-32767` range, enter the following command:
+
[source,terminal]
----
$ oc get svc
----
+
In the following example output, the updated port is `30327`:
+
.Example output
[source,terminal]
----
NAME    TYPE       CLUSTER-IP      EXTERNAL-IP   PORT(S)          AGE
httpd   NodePort   172.xx.xx.xx    <none>        8443:30327/TCP   109s
----
```
Feel free to spike it.
